From: "Dr. David Alan Gilbert" <linux@treblig.org>
Subject: lib/xxhash: comment out unused functions
Date: Mon, 14 Jul 2025 01:09:27 +0100

xxh32_digest() and xxh32_update() were added in 2017 in the original
xxhash commit, but have remained unused.

While I've mostly been deleting unused functions, this is a general
library and I see erofs is using other bits of xxh32, so it didn't seem
right just to delete them.

Comment them out with #if 0.
(Which checkpatch rightly warns about)

Link: https://lkml.kernel.org/r/20250714000927.294767-1-linux@treblig.org
Signed-off-by: Dr. David Alan Gilbert <linux@treblig.org>
Cc: Nick Terrell <terrelln@fb.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/xxhash.h |    4 ++++
 lib/xxhash.c           |    2 ++
 2 files changed, 6 insertions(+)

--- a/include/linux/xxhash.h~lib-xxhash-comment-out-unused-functions
+++ a/include/linux/xxhash.h
@@ -177,6 +177,9 @@ struct xxh64_state {
  */
 void xxh32_reset(struct xxh32_state *state, uint32_t seed);
 
+#if 0
+/* Currently unused, feel free to uncomment them if you use them. */
+
 /**
  * xxh32_update() - hash the data given and update the xxh32 state
  *
@@ -202,6 +205,7 @@ int xxh32_update(struct xxh32_state *sta
  * Return: The xxh32 hash stored in the state.
  */
 uint32_t xxh32_digest(const struct xxh32_state *state);
+#endif
 
 /**
  * xxh64_reset() - reset the xxh64 state to start a new hashing operation
--- a/lib/xxhash.c~lib-xxhash-comment-out-unused-functions
+++ a/lib/xxhash.c
@@ -267,6 +267,7 @@ void xxh64_reset(struct xxh64_state *sta
 }
 EXPORT_SYMBOL(xxh64_reset);
 
+#if 0
 int xxh32_update(struct xxh32_state *state, const void *input, const size_t len)
 {
 	const uint8_t *p = (const uint8_t *)input;
@@ -373,6 +374,7 @@ uint32_t xxh32_digest(const struct xxh32
 	return h32;
 }
 EXPORT_SYMBOL(xxh32_digest);
+#endif
 
 int xxh64_update(struct xxh64_state *state, const void *input, const size_t len)
 {
_
